tests/test_credentials.py

```diff
@@ -44,3 +49,32 @@ def test_minio_credentials_get_boto3_session():
 def test_credentials_get_client(credentials, client_type):
     with mock_s3():
         assert isinstance(credentials.get_client(client_type), BaseClient)
+
+
+@pytest.mark.parametrize("client_type", [member.value for member in ClientType])
+def test_get_client_cached(client_type):
```

Comment:
Can you add a test that ensures that changing some configuration on an `AwsCredentials` instance after instantiation and fetching a client causes a cache miss?